### PR TITLE
Added YAML exception handling to the file deserialization method

### PR DESCRIPTION
**Pull request by artv15**
My friend had some issues while creating custom roles, and we didn't exactly know what files had invalid YAML syntax, so I added exception handling for file deserialization method.

### CODE REVIEW

File: .vs/UncomplicatedCustomRoles2/v17/.wsuo
Comment (FoxWorn3365) [inline .vs/UncomplicatedCustomRoles2/v17/.wsuo:1]: you should delete the whole `.vs` folder

File: .vs/VSWorkspaceState.json
Comment (FoxWorn3365) [inline .vs/VSWorkspaceState.json:1]: you should delete the whole `.vs` folder

File: UncomplicatedCustomRoles/Manager/FileConfigs.cs [hunk 1]
```diff
@@ -27,19 +27,26 @@ public void LoadAll(string LocalDir = "")
...
+                        SpawnManager.RegisterCustomSubclass(SpawnManager.RenderExportMethodToInternal(Role));
+                    }
                 }
-                Dictionary<string, List<ExternalCustomRole>> Deserialized = Deserializer.Deserialize<Dictionary<string, List<ExternalCustomRole>>>(File.ReadAllText(FileName));
-                foreach (ExternalCustomRole Role in Deserialized["custom_roles"])
+                catch (YamlDotNet.Core.YamlException ex)
                 {
-                    Log.Debug($"Proposed to the registerer the external role {Role.Id} [{Role.Name}] from file:\n{FileName}");
-                    SpawnManager.RegisterCustomSubclass(SpawnManager.RenderExportMethodToInternal(Role));
+                    Log.Error($"Failed to parse {FileName} at {ex.End}. Recheck yml file's syntax and try again");
```
Comment (FoxWorn3365) [inline UncomplicatedCustomRoles/Manager/FileConfigs.cs:49]: should we show the `YamlException` message (should be `YamlException.message`)?

**Review by FoxWorn3365 — CHANGES_REQUESTED** [review]
Only a few changes are required, beside that this pr is good to go (and yeah I should have done that before :/ )

**Comment by artv15** [comment]
Removed .vs folder, but I believe that ex.Message would spew something like this, so I opted out to use ex.End instead

> (Line: 20, Col: 5, Idx: 369) - (Line: 20, Col: 5, Idx: 369): Exception during deserialization ---> System.InvalidOperationException: Failed to create an instance of type 'System.String'. ---> System.MissingMethodException: Default constructor not found for type System.String

**Comment by FoxWorn3365** [comment]
> Removed .vs folder, but I believe that ex.Message would spew something like this, so I opted out to use ex.End instead
> 
> > (Line: 20, Col: 5, Idx: 369) - (Line: 20, Col: 5, Idx: 369): Exception during deserialization ---> System.InvalidOperationException: Failed to create an instance of type 'System.String'. ---> System.MissingMethodException: Default constructor not found for type System.String

Hmm, if the stack trace is not in the `Message` why not adding it?

**Comment by artv15** [comment]
I don't think it would be directly helpful to the server owner trying to de-bug their .yml files.
Stack trace could be logged if debug is set to true though

**Comment by artv15** [comment]
whoops

**Comment by artv15** [comment]
while stack trace does look pretty bulky, it is only enabled if debug is set to true
debug log output
if it is disabled, it does look pretty clear what's wrong to me
normal log output

**Review by FoxWorn3365 — APPROVED** [review]

**Pull request merged**